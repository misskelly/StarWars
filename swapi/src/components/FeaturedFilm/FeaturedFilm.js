import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css';
import PropTypes from 'prop-types'

const FeaturedFilm = ({film}) => {

  return (
    <section className='film-container'>
      <Crawl
        title={film.title}
        subTitle={film.subtitle}
        text={film.scroll}
      />
      <article className='film-info'>
        <h3 className='film-title'>{film.title}</h3>
        <h4 className='subtitle'>{film.subtitle}</h4>
        <h5 className='release-year'>{film.year}</h5>
      </article>
      
    </section>
  )
}

FeaturedFilm.propTypes = {
  film: PropTypes.object.isRequired,
}

export default FeaturedFilm