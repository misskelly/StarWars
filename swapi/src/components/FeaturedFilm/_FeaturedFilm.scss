.film-container {
  grid-area: film;
}

h3, h4, h5, {
  margin: 0;
}
.crawler {
  background: transparent;
}

.film-info {
  text-align: center;
}
.film-title {
  line-height: 1;
  font-family: $alien;
  margin-top: 3rem;
}