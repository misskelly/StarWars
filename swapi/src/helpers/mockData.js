export const mockCleanFilm = {
     scroll: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     year: "1999"
   }
export const mockDirtyFilm = {
     opening_crawl: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     release_date: "1999-07-03"
   }


