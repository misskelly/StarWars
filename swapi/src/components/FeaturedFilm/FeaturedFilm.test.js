import React from 'react';
import FeaturedFilm from '../FeaturedFilm/FeaturedFilm';
import { shallow } from 'enzyme';
import Crawl from 'react-star-wars-crawl';
import { mockCleanFilm } from '../../helpers/mockData.js';


describe('FeaturedFilm', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<FeaturedFilm film={mockCleanFilm} />);
  });
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  

  it('renders a Crawl component with the correct props', () => {
    expect(wrapper.find(Crawl).prop('title')).toEqual(mockCleanFilm.title)

    expect(wrapper.find(Crawl).prop('subTitle')).toEqual(mockCleanFilm.subtitle)

    expect(wrapper.find(Crawl).prop('text')).toEqual(mockCleanFilm.scroll);
 
  });
});


  // // on component did mount we call fetch film
  // it('should call fetchFilm on mount', () => {
  //   // mount component
  //   jest.mock('../../helpers/fetchcalls.js')
  //    expect(mockFetchFilm).toHaveBeenCalled();
  //     // mock out if fetch is called, what to return
      
  //     // assert that state was set with film
  //     // state updates with new film after cdm
  // })






//   describe('getFilm', () => {
//     it('should update movie state on success', async () => {
//       await wrapper.instance().getMovie();
//       expect(wrapper.state('film')).toHaveProperty(...Object.keys(mockFilm));
//     });

//     it('should update error state on fail', async () => {
//       window.fetch = jest.fn().mockImplementation(() =>
//         Promise.resolve({
//           ok: false
//         })
//       );
//       await wrapper.instance().getMovie();
//       expect(wrapper.state('error')).toEqual('Unable to fetch film');
//     });
//   });
// });