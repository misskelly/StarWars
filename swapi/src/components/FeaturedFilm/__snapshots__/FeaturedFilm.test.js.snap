// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedFilm matches the snapshot 1`] = `
<section
  className="film-container"
>
  <Crawl
    text="Turmoil has engulfed the
Galactic Republic. The taxation
of trade routes to outlying star
systems is in dispute.

Hoping to resolve the matter
with a blockade of deadly
battleships, the greedy â€¦"
    title="The Phantom Menace"
  />
  <article
    className="film-info"
  >
    <h3
      className="film-title"
    >
      The Phantom Menace
    </h3>
    <h4
      className="subtitle"
    />
    <h5
      className="release-year"
    >
      1999
    </h5>
  </article>
</section>
`;
